Vector.hpp

```diff
@@ -0,0 +1,246 @@
+#ifndef VECTOR_HPP
+
+#define VECTOR_HPP
+
+#include <iostream>
+
+
+template<typename T>
+class Vector
+{
+public:
+	/**
+	 * @brief Default constructor.
+	 *
+	 * Constructs an empty Vector with an initial capacity of 2.
+	 */
+	Vector()
+	{
+		Realloc(2);
+	}
+	/**
+	 * @brief Destructor.
+	 *
+	 * Frees the dynamically allocated memory.
+	 */
+	~Vector()
+	{
+
+		delete(_dynamicVector);
+	}
+
+	/*@brief Adds an element to the end of the vector.
+	*
+     **/
+	void pushBack(const T& value)
+		
+	{
+		if (_size >= _capacity)
+		{
+			Realloc(_capacity + _capacity / 2);
```

Comment:
once again you got yourself a magic number.
for clarity on mathematical equations i would suggest using `()`